### ISSUE TITLE
unexpected end-of-string in org.owasp.dependencycheck.analyzer.CPEAnalyzer.searchCPE

### ISSUE THREAD

**Issue by dietzsch** [dependency-check/dependency-check-gradle#219]
Today I tried to update a few projects the dependency check plugin from version 6.1.6 to 6.2.0. All the projects have worked previously fine, but with the latest version I have two different issues:

* in two build pipelines the dependency check plugin does not return. It runs for an hour till the build is canceled without any exception or message indicating an error.
* for other build pipelines I see an exception in org.owasp.dependencycheck.analyzer.CPEAnalyzer.searchCPE

An unexpected error occurred during analysis of '/home/jenkins/.gradle/caches/modules-2/files-2.1/mysql/mysql-connector-java/8.0.25/f8b9123acd13058c941aff25f308c9ed8000bb73/mysql-connector-java-8.0.25.jar' (CPE Analyzer): unexpected end-of-string

java.lang.IllegalArgumentException: unexpected end-of-string
	at org.apache.lucene.util.automaton.RegExp.next(RegExp.java:1118)
	at org.apache.lucene.util.automaton.RegExp.parseCharExp(RegExp.java:1265)
	at org.apache.lucene.util.automaton.RegExp.parseSimpleExp(RegExp.java:1260)
	at org.apache.lucene.util.automaton.RegExp.parseCharClassExp(RegExp.java:1192)
	at org.apache.lucene.util.automaton.RegExp.parseComplExp(RegExp.java:1180)
	at org.apache.lucene.util.automaton.RegExp.parseRepeatExp(RegExp.java:1149)
	at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1142)
	at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        ...
	at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
	at org.apache.lucene.util.automaton.RegExp.parseInterExp(RegExp.java:1135)
	at org.apache.lucene.util.automaton.RegExp.parseUnionExp(RegExp.java:1129)
	at org.apache.lucene.util.automaton.RegExp.<init>(RegExp.java:487)
	at org.apache.lucene.search.RegexpQuery.<init>(RegexpQuery.java:138)
	at org.apache.lucene.search.RegexpQuery.<init>(RegexpQuery.java:121)
	at org.apache.lucene.search.RegexpQuery.<init>(RegexpQuery.java:92)
	at org.apache.lucene.queryparser.classic.QueryParserBase.newRegexpQuery(QueryParserBase.java:578)
	at org.apache.lucene.queryparser.classic.QueryParserBase.getRegexpQuery(QueryParserBase.java:760)
	at org.apache.lucene.queryparser.classic.QueryParserBase.handleBareTokenQuery(QueryParserBase.java:826)
	at org.apache.lucene.queryparser.classic.QueryParser.Term(QueryParser.java:469)
	at org.apache.lucene.queryparser.classic.QueryParser.Clause(QueryParser.java:355)
	at org.apache.lucene.queryparser.classic.QueryParser.Query(QueryParser.java:303)
	at org.apache.lucene.queryparser.classic.QueryParser.Clause(QueryParser.java:359)
	at org.apache.lucene.queryparser.classic.QueryParser.Query(QueryParser.java:244)
	at org.apache.lucene.queryparser.classic.QueryParser.TopLevelQuery(QueryParser.java:215)
	at org.apache.lucene.queryparser.classic.QueryParserBase.parse(QueryParserBase.java:109)
	at org.owasp.dependencycheck.data.cpe.AbstractMemoryIndex.parseQuery(AbstractMemoryIndex.java:277)
	at org.owasp.dependencycheck.analyzer.CPEAnalyzer.searchCPE(CPEAnalyzer.java:446)
	at org.owasp.dependencycheck.analyzer.CPEAnalyzer.determineCPE(CPEAnalyzer.java:280)
	at org.owasp.dependencycheck.analyzer.CPEAnalyzer.analyzeDependency(CPEAnalyzer.java:785)
	at org.owasp.dependencycheck.analyzer.AbstractAnalyzer.analyze(AbstractAnalyzer.java:131)
	at org.owasp.dependencycheck.AnalysisTask.call(AnalysisTask.java:88)
	at org.owasp.dependencycheck.AnalysisTask.call(AnalysisTask.java:37)

Both issues seem to be related to the new version, as there was no other change to the build or source code.

**Comment by jshiell** [comment]
I'm also seeing this on upgrade, again with MySQL connector:

```
An unexpected error occurred during analysis of '/Users/jsl0346/.gradle/caches/modules-2/files-2.1/mysql/mysql-connector-java/8.0.22/ec8bfce16c8bd18d6cb598b1147aa145be1408f/mysql-connector-java-8.0.22.jar' (CPE Analyzer): unexpected end-of-string

java.lang.IllegalArgumentException: unexpected end-of-string
        at org.apache.lucene.util.automaton.RegExp.next(RegExp.java:1118)
        at org.apache.lucene.util.automaton.RegExp.parseCharExp(RegExp.java:1265)
        at org.apache.lucene.util.automaton.RegExp.parseSimpleExp(RegExp.java:1260)
        at org.apache.lucene.util.automaton.RegExp.parseCharClassExp(RegExp.java:1192)
        at org.apache.lucene.util.automaton.RegExp.parseComplExp(RegExp.java:1180)
        at org.apache.lucene.util.automaton.RegExp.parseRepeatExp(RegExp.java:1149)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1142)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseConcatExp(RegExp.java:1144)
        at org.apache.lucene.util.automaton.RegExp.parseInterExp(RegExp.java:1135)
        at org.apache.lucene.util.automaton.RegExp.parseUnionExp(RegExp.java:1129)
        at org.apache.lucene.util.automaton.RegExp.<init>(RegExp.java:487)
        at org.apache.lucene.search.RegexpQuery.<init>(RegexpQuery.java:138)
        at org.apache.lucene.search.RegexpQuery.<init>(RegexpQuery.java:121)
        at org.apache.lucene.search.RegexpQuery.<init>(RegexpQuery.java:92)
        at org.apache.lucene.queryparser.classic.QueryParserBase.newRegexpQuery(QueryParserBase.java:578)
        at org.apache.lucene.queryparser.classic.QueryParserBase.getRegexpQuery(QueryParserBase.java:760)
        at org.apache.lucene.queryparser.classic.QueryParserBase.handleBareTokenQuery(QueryParserBase.java:826)
        at org.apache.lucene.queryparser.classic.QueryParser.Term(QueryParser.java:469)
        at org.apache.lucene.queryparser.classic.QueryParser.Clause(QueryParser.java:355)
        at org.apache.lucene.queryparser.classic.QueryParser.Query(QueryParser.java:303)
        at org.apache.lucene.queryparser.classic.QueryParser.Clause(QueryParser.java:359)
        at org.apache.lucene.queryparser.classic.QueryParser.Query(QueryParser.java:244)
        at org.apache.lucene.queryparser.classic.QueryParser.TopLevelQuery(QueryParser.java:215)
        at org.apache.lucene.queryparser.classic.QueryParserBase.parse(QueryParserBase.java:109)
        at org.owasp.dependencycheck.data.cpe.AbstractMemoryIndex.parseQuery(AbstractMemoryIndex.java:277)
        at org.owasp.dependencycheck.analyzer.CPEAnalyzer.searchCPE(CPEAnalyzer.java:446)
        at org.owasp.dependencycheck.analyzer.CPEAnalyzer.determineCPE(CPEAnalyzer.java:280)
        at org.owasp.dependencycheck.analyzer.CPEAnalyzer.analyzeDependency(CPEAnalyzer.java:785)
        at org.owasp.dependencycheck.analyzer.AbstractAnalyzer.analyze(AbstractAnalyzer.java:131)
        at org.owasp.dependencycheck.AnalysisTask.call(AnalysisTask.java:88)
        at org.owasp.dependencycheck.AnalysisTask.call(AnalysisTask.java:37)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
        at java.base/java.lang.Thread.run(Thread.java:831)
```

The MySQL part of our suppressions file is:
```
    <suppress>
        <notes><![CDATA[
        Connector/J is not MySQL server.
        ]]></notes>
        <gav regex="true">^mysql:mysql-connector-java:.*$</gav>
        <cpe>cpe:/a:mysql:mysql</cpe>
        <cpe>cpe:/a:oracle:mysql</cpe>
    </suppress>
    ```

**Comment by jeremylong** [comment]
We just released the 6.2.1. The unexpected end of string was fixed, I believe in 6.2.0. The process just hanging *might* be fixed in 6.2.1 - we identified one problem that caused the process to have connection issues. However, there still might be an ongoing issue that is being tracked here: https://github.com/jeremylong/DependencyCheck/issues/3408

**Comment by dietzsch** [comment]
Hi @jeremylong, regarding the IllegalArgumentException I can confirm that this is resolved with the new version 6.2.1. On the other hand I have still projects where the dependencyCheckAggregate tasks hangs without any error message. Is the any specific logs I can collect to troubleshoot this issue? Thx!

**Comment by jshiell** [comment]
`IllegalArgumentException` is fixed here - thanks very much.
